scripts/fundTokenLocks.js

```diff
@@ -84,22 +74,36 @@ async function main () {
 
   // loop through each array of token locks and fund
   for (const lock of actions.locks) {
+    // Connect to lock for fund() execution
+    const TokenLock = await hre.ethers.getContractFactory('TokenLock');
+    const tokenlock = TokenLock.attach(lock.address);
     // extract correct arguments array from objects in actions.locks elements
     console.log('Processing the following lock contract:\n', lock);
-    let totalLockFund = toBN('0');
-    for (const entry of lock.parameters[6]) {
-      // keep a running total of the sum of the amounts locked
-      totalLockFund = totalLockFund.add(toBN(entry));
+    // check if tokenlock is already funded
+    const contractState = await tokenlock.state();
+    console.log('Contract state: ' + LockString[contractState]);
+    if (LockString[contractState] === 'Funded') {
+      console.error('Contract ' + lock.name + ' is already funded');
+    } else {
+      console.log('Contract ' + lock.name + ' is not funded, so proceeding to fund');
+      // TODO: move this into ./utils/total.js and handle different key format from deployTokenLocks
+      let totalLockFund = toBN('0');
+      for (const entry of lock.parameters[6]) {
```

Comment:
can we check the balance first and send only what remains to be sent? do not assume someone didn't send stuff before. ie. the script can fail after sending but before or during calling fund()

Reply:
rudolfix: "reviewed. i have just one more issue: we should check the balance of the Lock contract before sending WFAIR. and send only what's missing OR to be on the safe side, you can do approve instead of transfer. then the contract will do all the calculations and execute transferFrom maybe we should switch to approve?"

Reply:
I'll switch to approve

Reply:
It was easier just to check the balance rather than write an approves module like the transfers module